docs_src/design/ucr/Global Common Configuration.md

```diff
@@ -0,0 +1,169 @@
+
+### Existing solutions
+There are no existing solutions for global configuration for EdgeX since the current configuration implementation is specific to EdgeX.
+
+### Requirements
+
+#### General
+
+- Services shall be able reference a global common configuration in a manner that is flexible for use with and without the Service Provider
+- Common environment overrides shall be applied to global common configuration settings only once before the services receive the setting values
+
+    - This avoids the need for duplicate environment overrides on each service in the docker compose files
+
+- Services must be able to override any of the global common configuration settings with local service specific TOML values
+
+    !!! example - "Example Core Data specific **Writable.Telemetry** and **Service** configuration settings in local TOML file"
+        ```toml
+          [Writable.Telemetry]
+            [Writable.Telemetry.Metrics] # All service's metric names must be present in this list.
+            EventsPersisted = false
+            ReadingsPersisted = false
+
+        ...
+        
+        [Service]
+        Port = 59880
+        StartupMsg = "This is the Core Data Microservice"
+        ```
+
+- Environment variable overrides shall be applied in the same manner as they are today. 
+    - After loaded from file (prior to pushing into the the Configuration Provider) 
+    - After loaded from the Configuration Provider
+
+#### With Configuration Provider
+
+- Global common configuration shall be pre-loaded into the Configuration Provider where it is pulled from by each service.
+    - How and where the configuration is loaded into the Configuration Provider is left to design
+- Pre-loading of global common configuration must have a mechanism to indicate that the configuration pre-load is complete 
+- Each service must wait until the global common configuration pre-load is complete.
+- Each service's full configuration, global common and local TOML overrides, shall be pushed into the Configuration Provider
+    - Once initially loaded the service's full configuration is solely referenced from the Configuration Provider as it is today
+- The services shall be notified when a setting in the **Writeable** section of global common configuration has changed. Each service shall consume the updated setting into its on `Writable` section and push the update back into the Configuration Provider under the service's own full configuration
+- On subsequent startups each service shall load the global common configuration from the Configuration Provider and merge it with the service's previous complete configuration from the Configuration Provider and push resulting configuration back to service's full configuration in the Configuration Provider
+
+#### Without Configuration Provider
+
+- Services shall have some manner to load a global common configuration TOML file via a URI

```

Comment:
Good document @lenny-intel .  A couple of questions that might need clarification from a requirements perspective, but otherwise I think you have captured it.

Reply:
> or to entirely remove global configuration and defer entirely to services??

Yes, if global URI is not specified, it would be as it is today.

Reply:
> Good document @lenny-intel . A couple of questions that might need clarification from a requirements perspective, but otherwise I think you have captured it.

Thanks! This was meant to get the ideas rolling. Next step is to have a real-time discussion in one of the meetings, before December 5th. ;-)